storage/myisam/ha_myisam.cc

```diff
@@ -813,6 +813,10 @@ int ha_myisam::write_row(uchar *buf)
 {
   ha_statistic_increment(&SSV::ha_write_count);
 
+  DBUG_EXECUTE_IF("myisam_simulate_crash_in_write_row", {
```

Comment:
that's not entirely accurate. this issue manifests for any error in write_row, as long as `fill_table_stats()` was called prior.  It's not necessary for max_fs to be exceeded.  Internally, we've run into this deadlock without that being the case.